sw/simulation/agent.h

```diff
@@ -29,13 +29,14 @@ class Agent
   uint16_t ID; // ID of agent
   bool activated;
   float dt;
-  std::vector<float> state; // State vector
+  std::vector<float> state,laser_ranges; // State vector
```

Comment:
The laser is a sensor. You can move this to the omniscient_observer class, which simulates all the sensors such as range sensors ect.